core/services/ocr2/plugins/ocr2vrf/fixed_blockhash_provider.go

```diff
@@ -0,0 +1,109 @@
+package ocr2vrf
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/smartcontractkit/ocr2vrf/types"
+
+	"github.com/smartcontractkit/chainlink/core/chains/evm/client"
+)
+
+// Implementation of Blockhashes interface that returns blockhashes with fixed-sized windows relative to current block height
+type FixedBlockhashProvider struct {
+	client client.Client
+	// end block = current head - waitBlocks
+	waitBlocks uint
+	// start block = current head - lookbackBlocks
+	lookbackBlocks uint
+	// number of blocks to query in a batch
+	batchSize uint
+}
+
+type Block struct {
+	Number    int64
+	Hash      common.Hash
+	Timestamp time.Time
+}
+
+var _ types.Blockhashes = (*FixedBlockhashProvider)(nil)
+
+func NewFixedBlockhashProvider(client client.Client, waitBlocks, lookbackBlocks, batchSize uint) *FixedBlockhashProvider {
```

Comment:
Better encapsulation if we return `types.Blockhashes` here instead of the concreted type `*FixedBlockhashProvider`.